python/src/MetropolisHastingsImplementation_doc.i.in

```diff
@@ -0,0 +1,499 @@
+%define OT_MetropolisHastings_doc
+"Base class for Metropolis-Hastings algorithms.
+
+Available constructor:
+    MetropolisHastingsImplementation(*targetDistribution, initialState, marginalIndices*)
+
+    MetropolisHastingsImplementation(*targetLogPDF, support, initialState, marginalIndices*)
+
+Parameters
+----------
+targetDistribution : :class:`~openturns.Distribution`
+    Target distribution sampled
+targetLogPDF : :class:`~openturns.Function`
+    Target log-pdf as a function
+support : :class:`~openturns.Interval`
+    Support of the target when defined with *targetLogPDF*
+initialState : sequence of float
+    Initial state of the chain
+marginalIndices : sequence of int, optional
+    Bloc indices to indicate which components are updated
+
+Notes
+-----
+MetropolisHastingsImplementation provides a implementation of the concept of sampler, using a Monte-Carlo
+Markov Chain (MetropolisHastingsImplementation) algorithm starting from *initialState*. More precisely,
+let :math:`t(.)` be the PDF of its target distribution and :math:`d_{\theta}`
+its dimension, :math:`\pi(.)` be the PDF of the *prior* distribution,
+:math:`f(.|\vect{w})` be the PDF of the *conditional* distribution
+when its parameters are set to :math:`\vect{w}`, :math:`d_w` be the number of
+scalar parameters of *conditional* distribution (which corresponds to the
+dimension of the above :math:`\vect{w}`), :math:`g(.)` be the function
+corresponding to model and :math:`(\vect{y}^1, \dots, \vect{y}^n)` be the
+sample *observations* (of size :math:`n`):
+
+In the first usage, it creates a sampler based on a MCMC algorithm whose target
```

Comment:
This description seems mis-leading: the first usage defines a prior distribution using a Distribution object, whereas the second allows to define the log prior density, together with the range of associated parameters. The likelihood and link function are added optionally through the setLikelihood method.